drivers/hmis/app/graphql/types/application/user_dashboard.rb

```diff
@@ -0,0 +1,70 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Types
+  class Application::UserDashboard < Types::BaseObject
+    # Type for resolving data needed on the HMIS user dashboard.
+    # Underlying object is Hmis::User, who should always be the currently logged-in user.
+    description 'Resolves everything that is needed on the user dashboard'
+
+    field :id, ID, null: false
+    field :user_dashboard_config, Types::Application::UserDashboardConfig, null: false
+    field :staff_assignments, HmisSchema::StaffAssignment.page_type, null: true
+    field :ce_referral_steps, HmisSchema::CeReferralStep.page_type, null: true
+
+    def user_dashboard_config
+      {
+        id: object.id,
+        show_staff_assignment: show_staff_assignment,
+        show_referrals: show_referrals,
+      }
+    end
+
+    def staff_assignments
+      object.staff_assignments.
+        viewable_by(current_user).
+        open_on_date. # This will include households that exited today
+        order(created_at: :desc, id: :desc)
+    end
+
+    def ce_referral_steps
```

Comment:
We still need to check whether show_referrals is true here, otherwise someone could resolve things they don't have access to (because the permission check doesn't occur on this field). Even though we know the frontend won't request it, we're still making it available in the API here.

```suggestion
    def ce_referral_steps
          return unless show_referrals
```

This could benefit from memoizing the private `show_` methods:

```
def show_referrals
  @show_referrals ||= begin
    return false unless Hmis::Ce.configuration.enabled?

    can_view_referrals = object.permissions?(:can_view_referrals, :can_view_own_referrals)
    can_perform_referral_steps = object.permissions?(:can_perform_any_referral_tasks, :can_perform_own_referral_tasks)

    can_view_referrals && can_perform_referral_steps
  end
end
```

And I would suggest using the same pattern for staff assignments, even though that one seems safe enough since it does check viewable_by, but it would be nice to have a consistent pattern to ensure we always check if the feature is enabled before resolving it.